app/models/manageiq/providers/hawkular/inventory/parser/middleware_manager.rb

```diff
@@ -144,44 +144,64 @@ def fetch_server_entities
+
+          props['Availability'] = availability.try(:[], 'data').try { first['value'] } || 'unknown'
+          props['Calculated Server State'] = props['Availability'] == 'up' ? props['Server State'] : props['Availability']
+        end
+      end
+
+      def fetch_availabilities_for(feeds, collection, metric_type_id)
         resources_by_metric_id = {}
         metric_id_by_resource_path = {}
-        collector.feeds.each do |feed|
-          deployment_status_mt_path = ::Hawkular::Inventory::CanonicalPath.new(
-            :metric_type_id => 'Deployment%20Status~Deployment%20Status', :feed_id => feed
-          )
-          deployment_status_metrics = collector.metrics_for_metric_type(deployment_status_mt_path)
-          deployment_status_metrics.each do |deployment_status_metric|
-            deployment_status_metric_path = ::Hawkular::Inventory::CanonicalPath.parse(deployment_status_metric.path)
+
+        feeds.each do |feed|
+          status_metrics = collector.metrics_for_metric_type(feed, metric_type_id)
+          status_metrics.each do |status_metric|
+            status_metric_path = ::Hawkular::Inventory::CanonicalPath.parse(status_metric.path)
             # By dropping metric_id from the canonical path we end up with the resource path
             resource_path = ::Hawkular::Inventory::CanonicalPath.new(
-              :tenant_id    => deployment_status_metric_path.tenant_id,
-              :feed_id      => deployment_status_metric_path.feed_id,
-              :resource_ids => deployment_status_metric_path.resource_ids
+              :tenant_id    => status_metric_path.tenant_id,
+              :feed_id      => status_metric_path.feed_id,
+              :resource_ids => status_metric_path.resource_ids
             )
-            metric_id_by_resource_path[resource_path.to_s] = deployment_status_metric.hawkular_metric_id
+            metric_id_by_resource_path[CGI.unescape(resource_path.to_s)] = status_metric.hawkular_metric_id
           end
         end
-        persister.middleware_deployments.each do |deployment|
-          # Mark default status for all deployments
-          deployment.status = process_availability
-          path = ::Hawkular::Inventory::CanonicalPath.parse(deployment.ems_ref)
-          # for subdeployments use it's parent deployment availability.
-          path = path.up if path.resource_ids.last.include? CGI.escape('/subdeployment=')
-          # Ensure consistency on keys (resource_path) used on metric_id_by_resource_path
-          path = ::Hawkular::Inventory::CanonicalPath.new(:tenant_id    => path.tenant_id,
-                                                          :feed_id      => path.feed_id,
-                                                          :resource_ids => path.resource_ids)
-          path = path.to_s
+
+        collection.each do |item|
+          yield item, nil
+
+          path = CGI.unescape(item.model_class.try(:resource_path_for_metrics, item) || item.manager_uuid)
```

Comment:
One detail that might be important is that CGI is not exactly related (meaning-wise) to what the PR is doing, and I think `URI.unescape` or `URI.decode` would be better. There are some tokens that parse differently on both methods, but I think it is worth a try to make the code less all-over-the-place.